terraform/modules/filecoin_node/lotus.sh

```diff
@@ -86,16 +86,31 @@ sudo --user="${NEW_USER}" -- \
   containrrr/watchtower \
   --include-stopped --revive-stopped --stop-timeout 120s --interval 600
 
-# Set-up  New Relic Agent For logs collection and Infrastruture Metrics
-curl -Ls https://download.newrelic.com/install/newrelic-cli/scripts/install.sh | bash && \
+if [ -n "${NEW_RELIC_API_KEY}" ] || [ -n "${NR_LICENSE_KEY}" ]; then
+  # Set-up  New Relic Agent For logs collection and Infrastruture Metrics
+  curl -Ls https://download.newrelic.com/install/newrelic-cli/scripts/install.sh | bash && \
   sudo  NEW_RELIC_API_KEY="${NEW_RELIC_API_KEY}" \
   NEW_RELIC_ACCOUNT_ID="${NEW_RELIC_ACCOUNT_ID}" \
   NEW_RELIC_REGION="${NEW_RELIC_REGION}" \
   /usr/local/bin/newrelic install -y
```

Comment:
I don't understand. Seems like you are assigning a variable to the same variable. Why is this needed?

Reply:
It is already set, given it is available as an environment variable. Could you please try running it without re-setting those variables?

Reply:
Why wouldn't it work? I get that this may be required if the variables would have different names, but they are precisely the same.

Reply:
So removing the assignments *will* work, right?

Reply:
Could you please print the error message, then?

Reply:
`NEW_RELIC_API_KEY` is already set. There is a reason to keep it the way it is, but it's not because, without this explicit assignment, it won't work.

Reply:
I don't mean running it separately, I mean running it in the same environment that is available when running this script and hitting this branch.

Reply:
Do you know why it doesn't work? It's good to have this kind of exercise.

Reply:
Let's not spend more time on it, perhaps. The crux is that the variables do not get re-exported because you are running the command as root without preserving the environmental variables with `-E`. That is to say, you can replace:
```
sudo  $NEW_RELIC_API_KEY \
  $NEW_RELIC_ACCOUNT_ID \
  $NEW_RELIC_REGION \
/usr/local/bin/newrelic install -y 
```
with `sudo -E /usr/local/bin/newrelic install -y`

You can verify that it works. There are no magical parameters.

That being said, I'm okay with keeping it as is in this form or (even better) with an additional comment on why this is needed.

Reply:
@samuelarogbonlo Please point me to a manual with a definition of an "input variable". If you could check the `sudo -E` I mentioned before it'd be great to clarify things.

Reply:
Ah. So this is what it was missing. This is not a real script, it's a template file. In such a case, a comment will certainly help to distinguish template substitution from environmental variables.

Reply:
That's wildly confusing. Hopefully, Terraform will resolve this. https://github.com/hashicorp/terraform/issues/28077